arrow-data/src/main/kotlin/arrow/data/Weak.kt

```diff
@@ -0,0 +1,82 @@
+package arrow.data
+
+import arrow.core.*
+import arrow.higherkind
+import java.lang.ref.WeakReference
```

Comment:
Could you please typealias WeakReference to WeakRef in a new object file called `Platform` inside a folder called internal. That way when we have to replace it it'll be more obvious.